protocol/communities/persistence.go

```diff
@@ -350,7 +339,7 @@ func (p *Persistence) JoinedAndPendingCommunitiesWithRequests(memberIdentity *ec
 }
 
 func (p *Persistence) DeletedCommunities(memberIdentity *ecdsa.PublicKey) (comms []*Community, err error) {
-	query := communitiesBaseQuery + ` WHERE NOT c.Joined AND (r.community_id IS NULL or r.state != ?)`
+	query := communitiesBaseQuery + ` WHERE NOT c.Joined AND NOT c.spectated AND r.state != ?`
```

Comment:
I think we should sync all not joined communities, also the spectated ones :thinking: 

Consider the case:
```
Join community A -> backup
Leave community A -> spectate community A -> backup (community A would not be backed-up)
```

Reply:
🤔... let's see. When we leave a community, we automatically stop spectating it.

In this branch :

```
Leave community A -> spectate community A -> backup (community A would not be backed-up)```

The `spectate community A` would be accomplished by the entire flow that involves pre-joining a community, but then the part of the query involved here is the pending joining a community.
Am I missing something here?

Reply:
I agree indeed. Sending another revision